### DISCUSSION TITLE
How to bundle precompiled binary and other files with my neu app ?

### DISCUSSION THREAD

**Discussion by gaurav-nelson** [neutralinojs/neutralinojs#807] [Q&A]
I am trying to build a frontend for a CLI tool, how can I bundle the pre-built binary and the configuration files along with my neu app ?

Is there a way already or can we add it as an enhancement?

PS: https://stackoverflow.com/questions/38361996/how-can-i-bundle-a-precompiled-binary-with-electron

**Comment by shalithasuranga** [comment]
Hi.. there can be several approaches to do this. But, the easiest way is to store binaries inside the `extensions` directory. So, the CLI's `neu build` command will copy `extensions` content to the application package. But, please note that Neutralinojs doesn't offer a way yet to add and load binaries to/from the `resources.neu` file.

**Comment by gaurav-nelson** [comment]
Thank you @shalithasuranga I tried to do that.
1. I enabled extensions in config.
2. Added a new extensions directory in my `src` folder.
3. Run the `neu build` command.

Which increased the size of `resources.neu` file, which means the binaries gets included. 
It is of no use unless I can use those resources in my app. Should I create a new GitHub issue for this enhancement?

**Reply by shalithasuranga** [reply]
Hey.. thanks for checking the solution. Please notice the following values in config.

- `resourcesPath` - Resources directory. All contents will be added to `resources.neu`
- `extensionsPath` - Extensions directory. Contents won't be added to `resources.neu`, but will be copied to the app build/release folder.

I think you've added binaries to a folder that resides in the `resourcesPath`. Please check our test app's directory structure/config: https://github.com/neutralinojs/neutralinojs/tree/main/bin

Yeah, sure you can suggest ideas with Github issues anytime :)
